src/loaders/OMETiffLoader.js

```diff
@@ -0,0 +1,322 @@
+import OMEXML from './omeXML';
+import { isInTileBounds } from './utils';
+
+// Credit to https://github.com/zbjornson/node-bswap/blob/master/bswap.js for the implementation.
+// I could not get this to import, and it doesn't appear anyone else can judging by the "Used by" on github.
+function flip16(info) {
+  const flipper = new Uint8Array(info.buffer, info.byteOffset, info.length * 2);
+  const len = flipper.length;
+  for (let i = 0; i < len; i += 2) {
+    const t = flipper[i];
+    flipper[i] = flipper[i + 1];
+    flipper[i + 1] = t;
+  }
+}
+
+function flip32(info) {
+  const flipper = new Uint8Array(info.buffer, info.byteOffset, info.length * 4);
+  const len = flipper.length;
+  for (let i = 0; i < len; i += 4) {
+    let t = flipper[i];
+    flipper[i] = flipper[i + 3];
+    flipper[i + 3] = t;
+    t = flipper[i + 1];
+    flipper[i + 1] = flipper[i + 2];
+    flipper[i + 2] = t;
+  }
+}
+
+function flip64(info) {
+  const flipper = new Uint8Array(info.buffer, info.byteOffset, info.length * 8);
+  const len = flipper.length;
+  for (let i = 0; i < len; i += 8) {
+    let t = flipper[i];
+    flipper[i] = flipper[i + 7];
+    flipper[i + 7] = t;
+    t = flipper[i + 1];
+    flipper[i + 1] = flipper[i + 6];
+    flipper[i + 6] = t;
+    t = flipper[i + 2];
+    flipper[i + 2] = flipper[i + 5];
+    flipper[i + 5] = t;
+    t = flipper[i + 3];
+    flipper[i + 3] = flipper[i + 4];
+    flipper[i + 4] = t;
+  }
+}
+
```

Comment:
can you explain what these are doing? why do the bytes need to be swapped?

Reply:
geotiff doesn't handle the endianness of the incoming information so we need to check it: https://github.com/hubmapconsortium/vitessce-image-viewer/pull/157/files#diff-aa25e815136ffd93913b888c4557329aR276